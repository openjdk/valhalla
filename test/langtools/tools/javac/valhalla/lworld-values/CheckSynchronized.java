/*
 * @test /nodynamiccopyright/
 * @summary Check behavior of synzhronized key word on value instances and methods.
 *
 * @compile/fail/ref=CheckSynchronized.out -XDrawDiagnostics CheckSynchronized.java
 */

primitive final class CheckSynchronized {
    synchronized void foo() { // <<-- ERROR, no monitor associated with `this'
    }
    void goo() {
        synchronized(this) {} // <<-- ERROR, no monitor associated with `this'
    }
    synchronized static void zoo(CheckSynchronized cs) { // OK, static method.
        synchronized(cs) {    // <<-- ERROR, no monitor associated with value instance.
        }
    }
    int x = 10;
}
