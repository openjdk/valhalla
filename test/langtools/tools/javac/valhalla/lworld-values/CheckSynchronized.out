CheckSynchronized.java:9:23: compiler.err.mod.not.allowed.here: synchronized
CheckSynchronized.java:12:9: compiler.err.type.found.req: CheckSynchronized, (compiler.misc.type.req.identity)
CheckSynchronized.java:15:9: compiler.err.type.found.req: CheckSynchronized, (compiler.misc.type.req.identity)
CheckSynchronized.java:19:9: compiler.err.type.found.req: CheckSynchronized.ref, (compiler.misc.type.req.identity)
CheckSynchronized.java:23:9: compiler.err.type.found.req: int, (compiler.misc.type.req.identity)
5 errors
